In [ ]:
# Load the necessary libraries
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
from matplotlib.colors import ListedColormap

# Load your dataset
df = pd.read_csv('../../dsi_team_22/data/raw/pharma.csv')

# 7. General Statistics


### Average (Mean) across all 36 countries
| column | average | unit |
|--------|---------|------|
| USD CAP | 295.05 | in USD per capita (using economy-wide PPPs) |
| Percentage Health Spending | 16.41 | % |
| Percentage GDP | 1.17 | % |

#### Supporting Python Code:

In [ ]:
# Identify Pharma Data Set Averages
avg_usd_cap = df['USD_CAP'].mean()
avg_pc_healthxp = df['PC_HEALTHXP'].mean()
avg_pc_gdp = df['PC_GDP'].mean()

# Display the results
print("Average USD CAP:",avg_usd_cap)
print("Average PC Health Spending:",avg_pc_healthxp)
print("Average Percentage GDP:",avg_pc_gdp)

## ??. Country Spending Overview 

| question | analysis |
|----------|----------|
| Which country has the highest total health expenditure? | United States (total spend: 4186292.78 USD millions) |
| Which country has the lowest total health expenditure? | Iceland (total spend: 3752.89 USD millions) |

### Total List of Countries ranked by Total Spend

| rank | country code | country name | total spend |
|------|--------------|--------------|-------------|
| 1 | USA | United States | 4186292.78 |
| 2 | JPN | Japan | 1602492.93 |
| 3 | DEU | Germany | 1188168.49 |
| 4 | FRA | France | 802298.94 |
| 5 | ITA | Italy | 754377.19 |
| 6 | MEX | Mexico | 451903.24 |
| 7 | CAN | Canada | 445157.79 |
| 8 | ESP | Spain | 436073.35 |
| 9 | KOR | South Korea | 413876.98 |
| 10 | GBR | United Kingdom | 243390.12 |
| 11 | AUS | Australia | 210562.01 |
| 12 | POL | Poland | 155643.45 |
| 13 | NLD | Netherlands | 147102.32 |
| 14 | BEL | Belgium | 126330.37 |
| 15 | GRC | Greece | 113206.42 |
| 16 | PRT | Portugal | 106881.27 |
| 17 | CHE | Switzerland | 103005.12 |
| 18 | SWE | Sweden | 93430.58 |
| 19 | AUT | Austria | 82723.22 |
| 20 | HUN | Hungary | 81580.59 |
| 21 | CZE | Czech Republic | 78489.23 |
| 22 | FIN | Finland | 52228.15 |
| 23 | IRL | Ireland | 46876.19 |
| 24 | RUS | Russia | 44655.62 |
| 25 | NOR | Norway | 44315.79 |
| 26 | DNK | Denmark | 38568.93 |
| 27 | SVK | Slovakia | 38472.30 |
| 28 | TUR | Turkey | 25138.33 |
| 29 | ISR | Israel | 16654.40 |
| 30 | LTU | Lithuania | 13859.94 |
| 31 | SVN | Slovenia | 12694.20 |
| 32 | NZL | New Zealand | 11907.15 |
| 33 | LVA | Latvia | 6829.89 |
| 34 | EST | Estonia | 5179.24 |
| 35 | LUX | Luxembourg | 4858.94 |
| 36 | ISL | Iceland | 3752.89 |

#### Supporting Python Code:

In [ ]:
# Identify the Top 10 highest spending countries
top_10_highest_spending = df.groupby('LOCATION')['TOTAL_SPEND'].sum().sort_values(ascending=False)

# Display the results
print("Top 10 Highest Spending Countries:")
print(top_10_highest_spending)

<a name="top10"></a>
### 4.1 What are the top 10 highest spending countries? 

| rank | country code | country name | total spend |
|------|--------------|--------------|-------------|
| 1 | USA | United States | 4186292.78 |
| 2 | JPN | Japan | 1602492.93 |
| 3 | DEU | Germany | 1188168.49 |
| 4 | FRA | France | 802298.94 |
| 5 | ITA | Italy | 754377.19 |
| 6 | MEX | Mexico | 451903.24 |
| 7 | CAN | Canada | 445157.79 |
| 8 | ESP | Spain | 436073.35 |
| 9 | KOR | South Korea | 413876.98 |
| 10 | GBR | United Kingdom | 243390.12 |

#### Supporting Python Code:


In [ ]:
# Identify the Top 10 highest spending countries
top_10_highest_spending = df.groupby('LOCATION')['TOTAL_SPEND'].sum().sort_values(ascending=False).head(10)

# Display the results
print("Top 10 Highest Spending Countries:")
print(top_10_highest_spending)

Top 10 Highest Spending Countries:
LOCATION
USA    4186292.78
JPN    1602492.93
DEU    1188168.49
FRA     802298.94
ITA     754377.19
MEX     451903.24
CAN     445157.79
ESP     436073.35
KOR     413876.98
GBR     243390.12
Name: TOTAL_SPEND, dtype: float64


<a name="midlevel"></a>

### 4.2 What are the mid level spending countries? 

| rank | country code | country name | total spend |
|------|--------------|--------------|-------------|
| 11 | AUS | Australia | 210562.01 |
| 12 | POL | Poland | 155643.45 |
| 13 | NLD | Netherlands | 147102.32 |
| 14 | BEL | Belgium | 126330.37 |
| 15 | GRC | Greece | 113206.42 |
| 16 | PRT | Portugal | 106881.27 |
| 17 | CHE | Switzerland | 103005.12 |
| 18 | SWE | Sweden | 93430.58 |
| 19 | AUT | Austria | 82723.22 |
| 20 | HUN | Hungary | 81580.59 |
| 21 | CZE | Czech Republic | 78489.23 |
| 22 | FIN | Finland | 52228.15 |
| 23 | IRL | Ireland | 46876.19 |
| 24 | RUS | Russia | 44655.62 |
| 25 | NOR | Norway | 44315.79 |
| 26 | DNK | Denmark | 38568.93 |

#### Supporting Python Code:

In [ ]:
# Calculate the total spend for each country
total_spend_by_country = df.groupby('LOCATION')['TOTAL_SPEND'].sum()

# Identify the top 10 and bottom 10 countries
top_10_countries = total_spend_by_country.nlargest(10)
bottom_10_countries = total_spend_by_country.nsmallest(10)

# Filter out the top 10 and bottom 10 countries
remaining_countries = total_spend_by_country.drop(top_10_countries.index).drop(bottom_10_countries.index).sort_values(ascending=False)

# Display the results
print("\nCountries Not in Top 10 or Bottom 10:")
print(remaining_countries)


Countries Not in Top 10 or Bottom 10:
LOCATION
AUS    210562.01
POL    155643.45
NLD    147102.32
BEL    126330.37
GRC    113206.42
PRT    106881.27
CHE    103005.12
SWE     93430.58
AUT     82723.22
HUN     81580.59
CZE     78489.23
FIN     52228.15
IRL     46876.19
RUS     44655.62
NOR     44315.79
DNK     38568.93
Name: TOTAL_SPEND, dtype: float64


<a name="bottom10"></a>

### 4.3 What are the bottom 10 lowest spending countries? 

| rank | country code | country name | total spend |
|------|--------------|--------------|-------------|
| 27 | SVK | Slovakia | 38472.30 |
| 28 | TUR | Turkey | 25138.33 |
| 29 | ISR | Israel | 16654.40 |
| 30 | LTU | Lithuania | 13859.94 |
| 31 | SVN | Slovenia | 12694.20 |
| 32 | NZL | New Zealand | 11907.15 |
| 33 | LVA | Latvia | 6829.89 |
| 34 | EST | Estonia | 5179.24 |
| 35 | LUX | Luxembourg | 4858.94 |
| 36 | ISL | Iceland | 3752.89 |

#### Supporting Python Code:

In [ ]:
# Identify the Bottom 10 lowest spending countries
bottom_10_lowest_spending = df.groupby('LOCATION')['TOTAL_SPEND'].sum().sort_values(ascending=True).head(10)

# Display the results
print("Bottom 10 Lowest Spending Countries:")
print(bottom_10_lowest_spending)

Bottom 10 Lowest Spending Countries:
LOCATION
ISL     3752.89
LUX     4858.94
EST     5179.24
LVA     6829.89
NZL    11907.15
SVN    12694.20
LTU    13859.94
ISR    16654.40
TUR    25138.33
SVK    38472.30
Name: TOTAL_SPEND, dtype: float64


## 5. Summary Statistics by Country Spending Category 

### Averages (Mean) per column

| category | % of health spending | % of GDP | USD per capita | total spend |
|----------|---------------------:|---------:|---------------:|------------:|
| top 10 countries | 717.68 | 50.70 | 13073.28 USD | 1052403.18 USD millions |
| mid-level countries | 474.55 | 34.59 | 9064.45 USD | 95349.95 USD millions |
| bottom 10 countries | 222.73 | 15.27 | 2990.46 USD | 13934.73 USD millions |

#### Supporting Python Code:

In [ ]:
# Calculate the total spend for each country
total_spend_by_country = df.groupby('LOCATION')['TOTAL_SPEND'].sum()

# Identify the top 10 and bottom 10 countries
top_10_total_spend = total_spend_by_country.nlargest(10)
bottom_10_total_spend = total_spend_by_country.nsmallest(10)

# Filter out the top 10 and bottom 10 countries
mid_level_total_spend = total_spend_by_country.drop(top_10_total_spend.index).drop(bottom_10_total_spend.index)

# Calculate the average total spend for each group
average_total_spend_top_10 = top_10_total_spend.mean()
average_total_spend_bottom_10 = bottom_10_total_spend.mean()
average_total_spend_mid_level = mid_level_total_spend.mean()

# Display the results
print("Average Total Spend of the Top 10 Countries:", average_total_spend_top_10)
print("Average Total Spend of the Mid-Level Countries:", average_total_spend_mid_level)
print("Average Total Spend of the Bottom 10 Countries:", average_total_spend_bottom_10)

Average Total Spend of the Top 10 Countries: 1052403.1809999999
Average Total Spend of the Mid-Level Countries: 95349.95375
Average Total Spend of the Bottom 10 Countries: 13934.728


In [ ]:
# Calculate the percentage of health spending for each country
pc_healthxp_by_country = df.groupby('LOCATION')['PC_HEALTHXP'].sum()

# Identify the top 10 and bottom 10 countries
top_10_pc_healthxp = pc_healthxp_by_country.nlargest(10)
bottom_10_pc_healthxp = pc_healthxp_by_country.nsmallest(10)

# Filter out the top 10 and bottom 10 countries
mid_level_pc_healthxp = pc_healthxp_by_country.drop(top_10_pc_healthxp.index).drop(bottom_10_pc_healthxp.index)

# Calculate the average total spend for each group
average_pc_healthxp_top_10 = top_10_pc_healthxp.mean()
average_pc_healthxp_bottom_10 = bottom_10_pc_healthxp.mean()
average_pc_healthxp_mid_level = mid_level_pc_healthxp.mean()

# Display the results
print("Average pc of health spending of the Top 10 Countries:", average_pc_healthxp_top_10)
print("Average pc of health spending of the Mid-Level Countries:", average_pc_healthxp_mid_level)
print("Average pc of health spending of the Bottom 10 Countries:", average_pc_healthxp_bottom_10)

Average pc of health spending of the Top 10 Countries: 717.6775
Average pc of health spending of the Mid-Level Countries: 474.55343750000003
Average pc of health spending of the Bottom 10 Countries: 222.7304


In [ ]:
# Calculate the percentage of GDP for each country
pc_gdp_by_country = df.groupby('LOCATION')['PC_GDP'].sum()

# Identify the top 10 and bottom 10 countries
top_10_pc_gdp = pc_gdp_by_country.nlargest(10)
bottom_10_pc_gdp = pc_gdp_by_country.nsmallest(10)

# Filter out the top 10 and bottom 10 countries
mid_level_pc_gdp = pc_gdp_by_country.drop(top_10_pc_gdp.index).drop(bottom_10_pc_gdp.index)

# Calculate the average total spend for each group
average_pc_gdp_top_10 = top_10_pc_gdp.mean()
average_pc_gdp_bottom_10 = bottom_10_pc_gdp.mean()
average_pc_gdp_mid_level = mid_level_pc_gdp.mean()

# Display the results
print("Average pc of gdp of the Top 10 Countries:", average_pc_gdp_top_10)
print("Average pc of gdp of the Mid-Level Countries:", average_pc_gdp_mid_level)
print("Average pc of gdp of the Bottom 10 Countries:", average_pc_gdp_bottom_10)

Average pc of gdp of the Top 10 Countries: 50.6995
Average pc of gdp of the Mid-Level Countries: 34.5884375
Average pc of gdp of the Bottom 10 Countries: 15.269199999999998


In [ ]:
# Calculate the USD per capita for each country
usd_cap_by_country = df.groupby('LOCATION')['USD_CAP'].sum()

# Identify the top 10 and bottom 10 countries
top_10_usd_cap = usd_cap_by_country.nlargest(10)
bottom_10_usd_cap = usd_cap_by_country.nsmallest(10)

# Filter out the top 10 and bottom 10 countries
mid_level_usd_cap = usd_cap_by_country.drop(top_10_usd_cap.index).drop(bottom_10_usd_cap.index)

# Calculate the average total spend for each group
average_usd_cap_top_10 = top_10_usd_cap.mean()
average_usd_cap_bottom_10 = bottom_10_usd_cap.mean()
average_usd_cap_mid_level = mid_level_usd_cap.mean()

# Display the results
print("Average USD per capita of the Top 10 Countries:", average_usd_cap_top_10)
print("Average USD per capita of the Mid-Level Countries:", average_usd_cap_mid_level)
print("Average USD per capita of the Bottom 10 Countries:", average_usd_cap_bottom_10)

Average USD per capita of the Top 10 Countries: 13073.2844
Average USD per capita of the Mid-Level Countries: 9064.449625000001
Average USD per capita of the Bottom 10 Countries: 2990.4642999999996


## 9. Addressing Inaccuracies

### Results are skewed by the number of observations
The output of "total_spend" for a country is the sum of all observations (rows) for that country in the dataset. If one country has 18 observations and another has 40, the TOTAL_SPEND for each country will be the sum of TOTAL_SPEND values from those 18 or 40 observations, respectively.

Meaning, the total spend will be influenced by the number of observations for each country. Countries with more observations will likely have higher total spend values simply because there are more data points contributing to the sum. This can skew the results if the number of observations varies significantly between countries. 

### Addressing the Skew
To address this potential skew, you might want to consider normalizing the spend values. Here are a few approaches you can take:

* Average Spend per Observation: Calculate the average spend per observation for each country. This can give a more balanced view, regardless of the number of observations.

* Median Spend: Use the median spend instead of the sum or mean, as the median is less sensitive to outliers and the number of observations.

#### Average Spend per Observation

| rank | country code | country name    | average spend per observation |
|------|--------------|-----------------|-------------------------------|
| 1    | USA          | United States   | 261643.298750                 |
| 2    | JPN          | Japan           | 50077.904062                  |
| 3    | RUS          | Russia          | 44655.620000                  |
| 4    | FRA          | France          | 26743.298000                  |
| 5    | MEX          | Mexico          | 26582.543529                  |
| 6    | DEU          | Germany         | 26403.744222                  |
| 7    | ITA          | Italy           | 26013.006552                  |
| 8    | ESP          | Spain           | 13627.292187                  |
| 9    | POL          | Poland          | 11117.389286                  |
| 10   | CAN          | Canada          | 9892.395333                   |
| 11   | KOR          | South Korea     | 8805.893191                   |
| 12   | GBR          | United Kingdom  | 7851.294194                   |
| 13   | AUS          | Australia       | 4785.500227                   |
| 14   | GRC          | Greece          | 4192.830370                   |
| 15   | HUN          | Hungary         | 3708.208636                   |
| 16   | NLD          | Netherlands     | 3420.984186                   |
| 17   | CHE          | Switzerland     | 3322.745806                   |
| 18   | AUT          | Austria         | 3181.662308                   |
| 19   | BEL          | Belgium         | 3081.228537                   |
| 20   | CZE          | Czech Republic  | 3018.816538                   |
| 21   | PRT          | Portugal        | 2740.545385                   |
| 22   | TUR          | Turkey          | 2285.302727                   |
| 23   | SVK          | Slovakia        | 2263.076471                   |
| 24   | ISR          | Israel          | 2081.800000                   |
| 25   | SWE          | Sweden          | 2031.099565                   |
| 26   | LTU          | Lithuania       | 1154.995000                   |
| 27   | IRL          | Ireland         | 1143.321707                   |
| 28   | FIN          | Finland         | 1135.394565                   |
| 29   | DNK          | Denmark         | 1071.359167                   |
| 30   | NOR          | Norway          | 942.889149                    |
| 31   | SVN          | Slovenia        | 906.728571                    |
| 32   | LVA          | Latvia          | 569.157500                    |
| 33   | NZL          | New Zealand     | 496.131250                    |
| 34   | EST          | Estonia         | 304.661176                    |
| 35   | LUX          | Luxembourg      | 231.378095                    |
| 36   | ISL          | Iceland         | 81.584565                     |


#### Supporting Python Code:

In [ ]:
average_spend_per_country = df.groupby('LOCATION')['TOTAL_SPEND'].mean().sort_values(ascending=False)

print("Average Total Spending Countries:")
print(average_spend_per_country)


Average Total Spending Countries:
LOCATION
USA    261643.298750
JPN     50077.904062
RUS     44655.620000
FRA     26743.298000
MEX     26582.543529
DEU     26403.744222
ITA     26013.006552
ESP     13627.292187
POL     11117.389286
CAN      9892.395333
KOR      8805.893191
GBR      7851.294194
AUS      4785.500227
GRC      4192.830370
HUN      3708.208636
NLD      3420.984186
CHE      3322.745806
AUT      3181.662308
BEL      3081.228537
CZE      3018.816538
PRT      2740.545385
TUR      2285.302727
SVK      2263.076471
ISR      2081.800000
SWE      2031.099565
LTU      1154.995000
IRL      1143.321707
FIN      1135.394565
DNK      1071.359167
NOR       942.889149
SVN       906.728571
LVA       569.157500
NZL       496.131250
EST       304.661176
LUX       231.378095
ISL        81.584565
Name: TOTAL_SPEND, dtype: float64


#### Median Spend

| rank | country code | country name    | median spend |
|------|--------------|-----------------|--------------|
| 1    | USA          | United States   | 275903.455   |
| 2    | RUS          | Russia          | 44655.620    |
| 3    | JPN          | Japan           | 40155.075    |
| 4    | MEX          | Mexico          | 31203.680    |
| 5    | ITA          | Italy           | 27984.090    |
| 6    | FRA          | France          | 27155.880    |
| 7    | DEU          | Germany         | 22996.320    |
| 8    | ESP          | Spain           | 12688.915    |
| 9    | POL          | Poland          | 11422.540    |
| 10   | CAN          | Canada          | 6974.070     |
| 11   | GBR          | United Kingdom  | 5334.670     |
| 12   | HUN          | Hungary         | 4010.575     |
| 13   | KOR          | South Korea     | 3729.280     |
| 14   | GRC          | Greece          | 3477.600     |
| 15   | AUT          | Austria         | 3250.675     |
| 16   | CZE          | Czech Republic  | 3076.195     |
| 17   | NLD          | Netherlands     | 2814.240     |
| 18   | AUS          | Australia       | 2764.915     |
| 19   | CHE          | Switzerland     | 2593.520     |
| 20   | PRT          | Portugal        | 2518.930     |
| 21   | SVK          | Slovakia        | 2401.340     |
| 22   | ISR          | Israel          | 2071.720     |
| 23   | BEL          | Belgium         | 2062.810     |
| 24   | SWE          | Sweden          | 1451.000     |
| 25   | LTU          | Lithuania       | 1131.030     |
| 26   | DNK          | Denmark         | 1009.555     |
| 27   | SVN          | Slovenia        | 922.460      |
| 28   | FIN          | Finland         | 851.320      |
| 29   | TUR          | Turkey          | 724.610      |
| 30   | NOR          | Norway          | 674.610      |
| 31   | LVA          | Latvia          | 588.680      |
| 32   | IRL          | Ireland         | 501.730      |
| 33   | NZL          | New Zealand     | 434.985      |
| 34   | EST          | Estonia         | 321.850      |
| 35   | LUX          | Luxembourg      | 243.370      |
| 36   | ISL          | Iceland         | 61.870       |

#### Supporting Python Code:

In [ ]:
median_spend_per_country = df.groupby('LOCATION')['TOTAL_SPEND'].median().sort_values(ascending=False)

print("Ranked Median Spending Countries:")
print(median_spend_per_country)


Ranked Median Spending Countries:
LOCATION
USA    275903.455
RUS     44655.620
JPN     40155.075
MEX     31203.680
ITA     27984.090
FRA     27155.880
DEU     22996.320
ESP     12688.915
POL     11422.540
CAN      6974.070
GBR      5334.670
HUN      4010.575
KOR      3729.280
GRC      3477.600
AUT      3250.675
CZE      3076.195
NLD      2814.240
AUS      2764.915
CHE      2593.520
PRT      2518.930
SVK      2401.340
ISR      2071.720
BEL      2062.810
SWE      1451.000
LTU      1131.030
DNK      1009.555
SVN       922.460
FIN       851.320
TUR       724.610
NOR       674.610
LVA       588.680
IRL       501.730
NZL       434.985
EST       321.850
LUX       243.370
ISL        61.870
Name: TOTAL_SPEND, dtype: float64
